api/src/test/java/org/pmiops/workbench/api/CohortBuilderControllerTest.java

```diff
@@ -1054,6 +1072,306 @@ public void findSurveyVersionByQuestionConceptId() {
+    assertTrue(exceptionThrown.getMessage().contains("Please provide a valid search term"));
+    // empty string length = 0
+    exceptionThrown =
+        assertThrows(
+            BadRequestException.class,
+            () -> controller.validateTerm(""),
+            "Expected BadRequestException is not thrown.");
+    assertTrue(exceptionThrown.getMessage().contains("Please provide a valid search term"));
+    // empty string length > 0
+    exceptionThrown =
+        assertThrows(
+            BadRequestException.class,
+            () -> controller.validateTerm("   "),
+            "Expected BadRequestException is not thrown.");
+    assertTrue(exceptionThrown.getMessage().contains("Please provide a valid search term"));
+  }
+
+  @Test
+  public void validateAgeTypeCaseSensitive() {
+    List<String> values =
+        Stream.of(AgeType.values()).map(AgeType::toString).collect(Collectors.toList());
+    for (String value : values) {
+      assertDoesNotThrow(
+          () -> controller.validateAgeType(value),
+          "BadRequestException is not expected to be thrown for [" + value + "]");
+    }
+    // expect exception for lowercase
+    values.replaceAll(String::toLowerCase);
+    for (String value : values) {
+      Throwable exceptionThrown =
+          assertThrows(
+              BadRequestException.class,
+              () -> controller.validateAgeType(value),
+              "Expected BadRequestException is not thrown.");
+      assertTrue(
+          exceptionThrown.getMessage().contains("Please provide a valid age type parameter"));
+    }
+  }
+
+  @Test
+  public void validateGenderOrSexTypeCaseSensitive() {
+    List<String> values =
+        Stream.of(GenderOrSexType.values())
+            .map(GenderOrSexType::toString)
+            .collect(Collectors.toList());
+    for (String value : values) {
+      assertDoesNotThrow(
+          () -> controller.validateGenderOrSexType(value),
+          "BadRequestException is not expected to be thrown for [" + value + "]");
+    }
+    // expect exception for lowercase
+    values.replaceAll(String::toLowerCase);
+    for (String value : values) {
+      Throwable exceptionThrown =
+          assertThrows(
+              BadRequestException.class,
+              () -> controller.validateGenderOrSexType(value),
+              "Expected BadRequestException is not thrown.");
+      assertTrue(
+          exceptionThrown
+              .getMessage()
+              .contains("Please provide a valid gender or sex at birth parameter"));
+    }
+  }
+
+  private String getDateRelativeToToday(int offsetYears, int offsetMonths, int offsetDate) {
+    SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+    Calendar today = Calendar.getInstance();
+    // set date of birth to today's month/date 25 years back
+    today.add(Calendar.YEAR, offsetYears);
+    today.add(Calendar.MONTH, offsetMonths);
+    today.add(Calendar.DATE, offsetDate);
+    return sdf.format(today.getTime());
+  }
```

Comment:
```suggestion
  private Date getDateRelativeToToday(int offsetYears, int offsetMonths, int offsetDate) {
    LocalDate today = LocalDate.now();
    return Date.valueOf(today.plusYears(offsetYears).plusMonths(offsetMonths).plusDays(offsetDate));
  }
```
I would use the Java 8 Date/Time API